agent/api/task_windows.go

```diff
@@ -69,8 +71,13 @@ func (task *Task) platformHostConfigOverride(hostConfig *docker.HostConfig) erro
 	// Convert the CPUShares to CPUPercent
 	hostConfig.CPUPercent = hostConfig.CPUShares * percentageFactor / int64(cpuShareScaleFactor)
 	if hostConfig.CPUPercent == 0 {
```

Comment:
Can you refactor this as `if hostConfig.CPUPercent == 0 && hostConfig.CPUShares != 0 `?